### PR TITLE
create a handler can list job details

### PR DESCRIPTION
**Pull request by lsy3993**
request is : 
```
curl -X POST -H "Content-Type: application/json" -d '{
        "name": "ccr_job_name"
    }' http://127.0.0.1:9190/job_detail
```

response is : 
```
{"success":true,"job detail":"{\"sync_type\":1,\"name\":\"ccr_test2\",
\"src\":{\"host\":\"xxx.xx.xx.xx\",
             \"port\":\"9030\",
             \"thrift_port\":\"9020\",
            \"is_master\":false,
            \"frontends\":[{\"host\":\"xxx.xx.xx.xx\",\"port\":\"9030\",\"thrift_port\":\"9020\",\"is_master\":true}],
            \"user\":\"root\",
            \"password\":\"\",
            \"cluster\":\"\",
            \"database\":\"test\",
             \"db_id\":10025,
              \"table\":\"table_a\",
              \"table_id\":10102},\  
"dest\":{\"host\":\"xxx.xx.xx.xx\",
              \"port\":\"9030\",
              \"thrift_port\":\"9020\",
              \"is_master\":false,
              \"frontends\":[{\"host\":\"xxx.xx.xx.xx\",\"port\":\"9030\",\"thrift_port\":\"9020\",\"is_master\":true}],
             \"user\":\"root\",
              \"password\":\"\",
              \"cluster\":\"\",
              \"database\":\"test\",
              \"db_id\":10025,
              \"table\":\"table_b\",
              \"table_id\":11411},
              \"skip_error\":false,
              \"state\":0}"}
```

### CODE REVIEW

**Comment by lsy3993** [comment]
@w41ter Is this response format correct?

File: pkg/service/http_service.go [hunk 1]
```diff
@@ -497,6 +497,55 @@ func (s *HttpService) listJobsHandler(w http.ResponseWriter, r *http.Request) {
 	}
 }
 
+// get job details
+func (s *HttpService) jobDetailHandler(w http.ResponseWriter, r *http.Request) {
+	log.Infof("get job detail")
+
+	type result struct {
+		*defaultResult
+		JobDetail string `json:"job detail"`
```
Comment (w41ter) [inline pkg/service/http_service.go]:
```suggestion
		JobDetail string `json:"job_detail"`
```

File: pkg/service/http_service.go [hunk 2]
```diff
@@ -497,6 +497,55 @@ func (s *HttpService) listJobsHandler(w http.ResponseWriter, r *http.Request) {
...
+
+	// Parse the JSON request body
+	var request CcrCommonRequest
+	err := json.NewDecoder(r.Body).Decode(&request)
+	if err != nil {
+		log.Warnf("get job detail failed: %+v", err)
+
+		jobResult = &result{
+			defaultResult: newErrorResult(err.Error()),
+		}
```
Comment (w41ter) [inline pkg/service/http_service.go:520]:
```suggestion
		}
		return
```

File: pkg/service/http_service.go [hunk 3]
```diff
@@ -497,6 +497,55 @@ func (s *HttpService) listJobsHandler(w http.ResponseWriter, r *http.Request) {
...
+	}
+
+	if request.Name == "" {
+		log.Warnf("get job detail failed: name is empty")
+
+		jobResult = &result{
+			defaultResult: newErrorResult(err.Error()),
```
Comment (w41ter) [inline pkg/service/http_service.go]:
```suggestion
			defaultResult: newErrorResult("name is empty"),
```

**Review by w41ter — CHANGES_REQUESTED** [review]

File: pkg/service/http_service.go [hunk 4]
```diff
@@ -497,6 +497,56 @@ func (s *HttpService) listJobsHandler(w http.ResponseWriter, r *http.Request) {
...
+		}
+		return
+	}
+
+	if request.Name == "" {
+		log.Warnf("get job detail failed: name is empty")
+
+		jobResult = &result{
+			defaultResult: newErrorResult("name is empty"),
+		}
```
Comment (w41ter) [inline pkg/service/http_service.go:529]:
```suggestion
		}
		return
```

**Review by w41ter — APPROVED** [review]

**Pull request merged**